admin/services/pupil-authentication.service.js

```diff
@@ -9,27 +9,26 @@ const pupilAuthenticationService = {
    * Authenticate a pupil from the school pin and pupil pin
    * @param pupilPin
    * @param schoolPin
-   * @return {Promise.<Pupil>}
+   * @return {object}
    */
   authenticate: async (pupilPin, schoolPin) => {
     const school = await schoolDataService.sqlFindOneBySchoolPin(schoolPin)
-    const pupil = await pupilDataService.findOne({
-      pin: pupilPin,
-      school: school && school.dfeNumber
-    })
-    if (!pupil || !school || !pinValidator.isActivePin(pupil.pin, pupil.pinExpiresAt)) {
+    if (!school) {
       throw new Error('Authentication failure')
     }
-    return pupil
+    const pupil = await pupilDataService.sqlFindOneByPinAndSchool(pupilPin, school.id)
+    if (!pupil || !pinValidator.isActivePin(pupil.pin, pupil.pinExpiresAt)) {
```

Comment:
Whilst it doesn't have to be done now, these two calls can consolidated into one call.  Perhaps worth adding a TODO at least?

Reply:
@jon-shipley todo included